File: src/api/expenses.ts
import { invoke } from "@tauri-apps/api/core";
import { TExpenses } from "../types";

export const addExpensesApi = async (data: TExpenses) => {
  try {
    const res = await invoke<TExpenses>("add_expenses", { data });
    return res;
  } catch (e) {
    console.error("Error adding expenses:", e);
    throw e;
  }
};

export const getExpensesByMonthApi = async (month: string) => {
  try {
    const res = await invoke<TExpenses>("get_expenses_by_month", { month });
    console.log("Expenses for month:", month, res);

    return res;
  } catch (e) {
    console.error("Error getting expenses:", e);
    throw e;
  }
};

export const updateExpensesByIdApi = async (id: string, data: TExpenses) => {
  try {
    const res = await invoke<boolean>("update_expenses_by_id", { id, data });
    return res;
  } catch (e) {
    console.error("Error updating expenses:", e);
    throw e;
  }
};

File: src-tauri/src/handlers/expenses.rs
use crate::datastore::expenses::{
    add_expenses_db, get_expenses_by_month_db, update_expenses_by_id_db, update_expenses_db,
};
use crate::types::Expenses;

#[tauri::command]
pub fn add_expenses(data: Expenses, window: tauri::Window) -> Result<String, String> {
    add_expenses_db(data, window)
}

#[tauri::command]
pub fn update_expenses(data: Expenses, window: tauri::Window) -> Result<String, String> {
    update_expenses_db(data, window)
}

#[tauri::command]
pub fn get_expenses_by_month(
    month: String,
    window: tauri::Window,
) -> Result<Option<Expenses>, String> {
    get_expenses_by_month_db(month, window)
}

#[tauri::command]
pub fn update_expenses_by_id(
    id: String,
    data: Expenses,
    window: tauri::Window,
) -> Result<bool, String> {
    update_expenses_by_id_db(id, data, window)
}
